hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/impl/TestHddsDispatcher.java

```diff
@@ -158,6 +170,86 @@ public void testContainerCloseActionWhenFull() throws IOException {
 
   }
 
+  @Test
+  public void testContainerCloseActionWhenVolumeFull() throws Exception {
+    String testDir = GenericTestUtils.getTempPath(
+        TestHddsDispatcher.class.getSimpleName());
+    OzoneConfiguration conf = new OzoneConfiguration();
+    conf.setStorageSize(HddsConfigKeys.HDDS_DATANODE_VOLUME_MIN_FREE_SPACE,
+        100.0, StorageUnit.BYTES);
+    DatanodeDetails dd = randomDatanodeDetails();
+
+    HddsVolume.Builder volumeBuilder =
+        new HddsVolume.Builder(testDir).datanodeUuid(dd.getUuidString())
+            .conf(conf).usageCheckFactory(MockSpaceUsageCheckFactory.NONE);
+    // state of cluster : capacity-used (140) > 100  ,datanode volume
+    // utilisation threshold not yet reached. container creates are successful.
```

Comment:
Is the scope of this PR restricted to cases when available space is equal or less than the configured minimum space? For example, what if the container's size was 139 bytes here?

Reply:
The PR only restricts container allocations when the volume approaches its capacity i.e its available space <= configured minimum, if even the available space is a byte greater than the configured minimum , container allocates would succeed. But after a while that byte would get filled up  and available would be less than the min space, this time during a chunk write to this container, it will realise this and ask SCM to close the container.